### DISCUSSION TITLE
SONOFF Zigbee 3.0 USB Dongle Plus doesn't work (Based on Texas Instruments CC2652/CC1352 chip)

### DISCUSSION THREAD

**Discussion by igtek** [Koenkk/zigbee2mqtt#10418] [Devices]
Hello, this is my first time in this community. I'm using Zigbee2MQTT from a few months and I consider it an extremely interesting and reliable tool.

I bought a more performance adapter to replace the classic CC2531 one.
I've done several tests with two different raspberry and restoring more times the image, upgrading node to different version, but all times with the same identical fault ("Error: SRSP - ZDO - mgmtPermitJoinReq after 6000ms"). I've also add an external powered USB hub.

Only sometimes the system runs correctly for few seconds or at most one-two minutes.

The adapter is been flashed with CC1352P2_CC2652P_launchpad_coordinator_20210708.hex, using SmartRF Flash Programmer 2 with successful result (erase, program and verify flagged).

The error is "Error: SRSP - ZDO - mgmtPermitJoinReq after 6000ms" or similar ones.

Here are the logs in debug mode:
Zigbee2MQTT:info  2021-12-28 23:49:58: Started frontend on port 0.0.0.0:8081
Zigbee2MQTT:debug 2021-12-28 23:50:19: Received MQTT message on 'zigbee2mqtt/bridge/request/permit_join' with data '{"device":null,"time":254,"transaction":"g7021-1","value":true}'
Zigbee2MQTT:info  2021-12-28 23:50:19: Zigbee: allowing new devices to join.
  zigbee-herdsman:adapter:zStack:znp:SREQ --> ZDO - mgmtPermitJoinReq - {"addrmode":15,"dstaddr":65532,"duration":254,"tcsignificance":0} +21s
  zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,5,37,54,15,252,255,254,0,228] +21s
Zigbee2MQTT:error 2021-12-28 23:50:25: Request 'zigbee2mqtt/bridge/request/permit_join' failed with error: 'SRSP - ZDO - mgmtPermitJoinReq after 6000ms'
Zigbee2MQTT:debug 2021-12-28 23:50:25: Error: SRSP - ZDO - mgmtPermitJoinReq after 6000ms
    at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
Zigbee2MQTT:info  2021-12-28 23:50:25: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{},"error":"SRSP - ZDO - mgmtPermitJoinReq after 6000ms","status":"error","transaction":"g7021-1"}'

System info:
linux 5.10.63-v7l+
node v. 16.13.1
npm v. 8.1.2
Zigbee2MQTT version 1.22.1

I attach the configuration.yaml and the log.txt files [configuration_log.zip](https://github.com/Koenkk/zigbee2mqtt/files/7786668/configuration_log.zip)

On the same hardware, if I use an adapter with CC2531, Zigbee2MQTT run properly.

Do you have any suggestions?

**Comment by igtek** [comment]
After paired a device successfully, the logs when an event occurs (example push a button) are as following:

Zigbee2MQTT:info  2021-12-29 21:33:32: MQTT publish: topic 'zigbee2mqtt/0x00124b0024fffb78', payload '{"battery":100,"voltage":3300}'
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [119,250] +59s
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [119,250] +1ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [247,248] +157ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [119,250,247,248] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [8,94,254] +36s
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [119,250,247,248,8,94,254] +1ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [8,70,252] +28s
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [254,8,70,252] +1ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [67,198,255] +848ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [254,8,70,252,67,198,255] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [66,199,255] +5s
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [254,8,70,252,67,198,255,66,199,255] +1ms
  zigbee-herdsman:adapter:zStack:unpi:parser <-- [66,66,255] +961ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [254,8,70,252,67,198,255,66,199,255,66,66,255] +0ms
  zigbee-herdsman:adapter:zStack:unpi:parser --> error Error: Invalid checksum
  zigbee-herdsman:adapter:zStack:unpi:parser     at Function.fromBuffer (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/unpi/frame.ts:48:19)
  zigbee-herdsman:adapter:zStack:unpi:parser     at Parser.parseNext (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/unpi/parser.ts:43:41)
  zigbee-herdsman:adapter:zStack:unpi:parser     at Parser._transform (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/unpi/parser.ts:19:14)
  zigbee-herdsman:adapter:zStack:unpi:parser     at Parser.Transform._write (node:internal/streams/transform:184:23)
  zigbee-herdsman:adapter:zStack:unpi:parser     at writeOrBuffer (node:internal/streams/writable:389:12)
  zigbee-herdsman:adapter:zStack:unpi:parser     at _write (node:internal/streams/writable:330:10)
  zigbee-herdsman:adapter:zStack:unpi:parser     at Parser.Writable.write (node:internal/streams/writable:334:10)
  zigbee-herdsman:adapter:zStack:unpi:parser     at SerialPort.ondata (node:internal/streams/readable:754:22)
  zigbee-herdsman:adapter:zStack:unpi:parser     at SerialPort.emit (node:events:390:28)
  zigbee-herdsman:adapter:zStack:unpi:parser     at addChunk (node:internal/streams/readable:315:12) +38ms
  zigbee-herdsman:adapter:zStack:unpi:parser --- parseNext [] +0ms

I'm believing that my adapter is faulty, but if I restart Z2M, it works for few seconds whithout need to unplug it. I hope somebody will be able to help me!
Is there any test to check the adapter? 
I tryed to flash again the adapter, but nothing change.

**Reply by Terminator0007** [reply]
Why you flash this Sonoff device ? Factory firmware is dedicated to work wit Zigbee2Mqtt ... I use this device and everything is ok. Problems are different on different systems. When I'm using Armbian on OrangePi Zero all is ok but many problems on DietPi .

**Reply by oh2xx** [reply]
I do not know about  @igtek but I am going to flash this one as required after changing the IEEEE address https://www.zigbee2mqtt.io/guide/adapters/flashing/copy_ieeaddr.html

**Reply by igtek** [reply]
I updated it because I was trying to find a solution to my problem. This is the first time that I'm using SONOFF Zigbee 3.0 USB Dongle Plus and so I've tried to do also that.
In fact, it doesn't need to reflash. 
It's needed, for example, if you want to set the ieee address because you want to replace another adapter without repairing all the devices.

**Reply by Hedda** [reply]
> Why you flash this Sonoff device ? Factory firmware is dedicated to work wit Zigbee2Mqtt 

While the firmware it ships with from factory do work most of the time it is highly recommended to upgrade firmware which not only fixes a lot of bugs from Texas Instruments upstream but also adds +9 dBm as the new transmission amplification. See discussions:

https://github.com/Koenkk/zigbee2mqtt/discussions/8840

and

https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/CHANGELOG.md

**Comment by igtek** [comment]
The issue is solved.

In my images there was a daemon of mine trying every minute to find a GSM modem pluggend in a USB port: it opens the serial port and writes a command on it and then closes it.
Obviously this was in conflict with Zigbee2mqtt.

The adapter "SONOFF Zigbee 3.0 USB Dongle Plus" works properly.
